kubeflow/hub/api/model_registry_client.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine, Iterator, Mapping
+import logging
+from typing import TYPE_CHECKING, Any, TypeVar
+
+from typing_extensions import Self
+
+if TYPE_CHECKING:
+    from model_registry import ModelRegistry
+    from model_registry.types import (
+        ModelArtifact,
+        ModelVersion,
+        RegisteredModel,
+        SupportedTypes,
+    )
+
+T = TypeVar("T")
+
+
+class ModelRegistryClient:
+    """Client for Kubeflow Model Registry operations.
+
+    Requires the model-registry package to be installed. Install it with:
+
+        pip install 'kubeflow[hub]'
+
+    """
+
+    def __init__(
+        self,
+        server_address: str,
+        port: int = 443,
+        *,
+        author: str,
+        is_secure: bool = True,
+        user_token: str | None = None,
+        user_token_envvar: str | None = None,
+        custom_ca: str | None = None,
+        custom_ca_envvar: str | None = None,
+        log_level: int = logging.WARNING,
+        async_runner: Callable[[Coroutine[Any, Any, T]], T] | None = None,
+    ):
```

Comment:
At the moment, this has the same `__init__` args as `ModelRegistry`, but thoughts on what to change would be great.

I'm not a huge fan of the keyword-only args, although it does offer flexibility to re-order args later without it being a breaking change.

I'm not sure we need all of these. For instance, `async_runner` can probably be removed at the least.